Pin black version, fix spotdb flake8 errors

slabasan opened the pull request:
Pin black version to maintain python 2.7 as a version that should be supported by Black's output. Newer versions removed support for python 2.7.

alfredo-gimenez commented on hatchet/caliper_reader.py:
```diff
@@ -0,0 +1,231 @@
+        # records is a list of lists, each inner list is a record
+        # containing both positional values and metrical values pertaining
+        # to one caliper sample (i.e. one time caliper iterrupted and recorded)
+        self.records = json_obj[self.records_key]
```
Can we avoid storing this in addition to creating the dataframe? It would make sense to run `create_cctree` here and pass in the records rather than keep both.

alfredo-gimenez replied:
I think I see why you did this now, looking at how you invoke `create_cctree`. I would suggest creating a separate cctree constructor using something like this: https://stackoverflow.com/a/2164383

alfredo-gimenez replied:
You could then have an accessor to get the tree rather than explicitly call `create_cctree`

alfredo-gimenez commented on hatchet/caliper_reader.py:
```diff
@@ -0,0 +1,231 @@
+        # column names, column metadata, records, and calling tree are stored in
+        # the json file file_name
+        with open(file_name) as fp:
+            json_obj = json.load(fp)
```
so, we've had trouble in the past using python to read files line by line. Pandas actually can do all of this pretty easily through `pd.from_json(filename, orient='look up the correct orientation')`. That should read in all the data much faster and you don't have to store raw records, then you can iterate over your dataframe instead of over `self.records` (also solves my other comment)

alfredo-gimenez commented on hatchet/caliper_reader.py:
```diff
@@ -0,0 +1,231 @@
+
+        # idx_has_record will be a set containing the list of json ccnode
+        # indices that have a record associated with them
+        idx_has_record = set()
```
if you have a "records" dataframe, you can just do `dataframe['idx'].unique()` to get the set of unique values in that column

matthewkotila replied:
my mistake, that variable never gets used after, it was an artifact from an older version of the file

slabasan committed:
pin black version for formatting python 2.7

slabasan committed:
spotdb flake8 errors

- add flake exception for pytest.importorskip of spotdb package
- install flake8-pytest-importorskip package for python 3.7

slabasan committed:
roundtrip: fix black errors, flake8 ignore init file

slabasan committed:
temporarily disable tests not passing python 3.8

slabasan committed:
pin coverage version for python 3.8
